src/main/java/com/rybka/view/ExchangeView.java

```diff
@@ -8,33 +12,34 @@
 
     private final Scanner scanner = new Scanner(System.in);
     private final ExchangeService service = new ExchangeService();
+    private final HibernateDAO hibernateDAO = new HibernateDAO();
 
     public void showDialog() {
 
         try {
             System.out.print("Insert base currency: ");
-            String userBaseCurrencyAbbreviation = scanner.next().toUpperCase();
+            var userBaseCurrency = scanner.next().toUpperCase();
             System.out.print("Enter value: ");
-            Double userValue = scanner.nextDouble();
+            var userValue = scanner.nextDouble();
             System.out.print("Insert target currency: ");
 
-            String userTargetCurrencyAbbreviation = scanner.next().toUpperCase();
+            var userTargetCurrency = scanner.next().toUpperCase();
 
-            service.loadCurrencyOf(userBaseCurrencyAbbreviation, userTargetCurrencyAbbreviation);
-            service.exchange(userValue);
-            service.getCurrencyObject();
+            var currencyObject = service.loadCurrencyOf(userBaseCurrency, userTargetCurrency);
+            var currency = service.calculateTotal(currencyObject, userValue);
+            hibernateDAO.save(currency);
```

Comment:
Currency
from, to, amount, rate, total, date

Reply:
fixed, but I use timestamp for exchange date